docs/resources/custom_domain_verification.html.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: "auth0"
+page_title: "Auth0: auth0_custom_domain_verification"
+description: |-
+  With this resource, you can verify a custom domain created with the `auth0_custom_domain` resource.
+---
+
+# auth0_custom_domain_verification
+
+With Auth0, you can use a custom domain to maintain a consistent user experience. This is a three-step process; you must configure the custom domain in Auth0, then create a DNS record for the domain, then verify the DNS record in Auth0. This resources allows for automating the verification part of the process.
+
+## Example Usage
+
+```hcl
+resource "auth0_custom_domain" "my_custom_domain" {
+  domain = "auth.example.com"
+  type = "auth0_managed_certs"
+  verification_method = "txt"
```

Comment:
Where is the documentation for those 2 inputs?

Reply:
According to the [api docs](https://auth0.com/docs/api/management/v2/#!/Custom_Domains/post_custom_domains), there's only one option: "txt". Maybe other types are just not documented

Reply:
Yeah that is what I thought too. I'm gonna open a support ticket and see what their response is.

Reply:
From Auth0

```
Good day,
Thank you for contacting support. I would be happy to assist.

Our documentation on how custom domains work as documented at https://auth0.com/docs/custom-domains#how-it-works notes that you have two choices when setting up a custom domain
1. use Auth0 Managed Certificates (https://auth0.com/docs/custom-domains#auth0-managed-certificates)
2. use a Self-managed Certificate (https://auth0.com/docs/custom-domains#self-managed-certificates).

When you are using Auth0 Managed Certificates, the CNAME record that you configure for your domain can be used to verify ownership as documented at https://auth0.com/docs/custom-domains/configure-custom-domains-with-auth0-managed-certificates#verify-ownership and complete the configuration of your custom domain.

With Self-managed Certificates, an option available to Enterprise subscribers, you are choosing to take ownership of more of the pipeline by managing the SSL certificate and setting up and managing the reverse proxy to send content to Auth0, with Auth0 negotiating SSL with the proxy and not the end-user client. As documented at https://auth0.com/docs/custom-domains/configure-custom-domains-with-self-managed-certificates#verify-ownership you would use the TXT verification information you noted as part of the steps for the process for verification in this use case.

When using the endpoint to create the custom domain that you referenced https://auth0.com/docs/api/management/v2/#!/Custom_Domains/post_custom_domains the parameter you have referenced is set when you plan to use a Self-managed Certificate as the method for configuring your custom domain, as it would be required for the verification process subsequently.

Please let me know if there is any additional assistance I can offer here.
```

Reply:
Thanks for following up on this! After fiddling a little, I am starting to believe the `verification_method` is not necessary at all. The Auth0 dashboard doesn't specify it at all, but relies on the type to chose the correct type. `auth0_managed_certs` default to `TXT` verification and `self_managed_certs` default to CNAME.

In fact, even in the test specifying `verification_method = "txt"` are ignored. The DNS entry created at DO is of CNAME type. 

Maybe best to deprecate `verification_method` altogether?

Reply:
I agree that since it depends on the `type` then better to remove it for less confusion.

Reply:
It would still be nice to have `type` documented to mention that it supports both `self_managed_certs` and `auth0_managed_certs`